src/map/utils/charutils.cpp

```diff
@@ -4470,15 +4471,30 @@ namespace charutils
             {
                 PChar->StatusEffectContainer->DelStatusEffect(EFFECT_DEDICATION);
             }
+        }
 
+        for (auto i = 2884; i <= 2892; ++i) // RHAPSODY KI are sequential, so start at WHITE and end at OCHRE
+        {
+            if (hasKeyItem(PChar, i))
+            {
+                bonus += 30;
+            }
+            else
+            {
+                break; // No need to check further as you can't get KI out of order, so break out.
+            }
         }
 
         bonus += (int32)(exp * (PChar->getMod(Mod::EXP_BONUS) / 100.0f));
```

Comment:
So if I'm reading this right, the ROV KI would only add 30 exp, instead of multiplying by 30%.

Reply:
```cpp
        auto RoVbonus = 0;
        for (auto i = 2884; i <= 2892; ++i) // RHAPSODY KI are sequential, so start at WHITE and end at OCHRE
        {
            if (hasKeyItem(PChar, i))
            {
                RoVbonus += 30;
```
```cpp
        bonus += (int32)(exp * (PChar->getMod(Mod::EXP_BONUS) + RoVbonus / 100.0f));
```
should fix that I think